Automattic/wp-calypso
# Checkout: Move lib/postal-code to wpcom-checkout package

#### Changes proposed in this Pull Request

This moves the entire `lib/postal-code` directory, that I think is only used by checkout, into the `@automattic/wpcom-checkout` package and converts it to TypeScript. The package has type safety and it will make it easier to migrate other parts of checkout to this package that rely on these functions. The functions have no dependencies of their own so this is fairly easy to move.

#### Testing instructions

There are two exports to verify:

The functionality of `tryToGuessPostalCodeFormat` (the main export of the library) is covered by unit tests, so the main thing to verify is that the imports have been moved to the package. To test `tryToGuessPostalCodeFormat`, make a purchase using checkout and nearly any payment method (except for PayPal). Verify there are no errors when submitting the purchase.

The other export of the package is `isValidPostalCode`. It is used by `validatePaymentDetails` and is therefore covered by the unit tests in `client/lib/checkout/test/validation.js`. There shouldn't be any need to test this manually.
